File: Linked List/ValidCircularLinkedList.js
function Node(value){
	this.value = value;
	this.next = null;
}

function validCirLL(node){
	var slow = node;
	var fast = node;
	while(fast.next.next != null && fast.next != null){
		slow = slow.next;
		fast = fast.next.next;
		if(slow.value === fast.value){
			return true;
		}
	}
	return false;
}

var a = new Node(1);
var b = new Node(2);
var c = new Node(3);
var d = new Node(4);

a.next = b;
b.next = c;
c.next = d;

var v = new Node(0);
var x = new Node(1);
var y = new Node(2);
var z = new Node(3);

v.next = x;
x.next = y;
y.next = z;
z.next = v;

console.log(validCirLL(a));
console.log(validCirLL(x));

File: Linked List/ValidCircularLinkedList.py
from nose.tools import assert_equal

class Node:
	def __init__(self, value):
		self.value = value
		self.next = None

def valid_circular_LL(node):
	slow = node
	fast = node
	while fast.next.next != None and fast.next != None:
		slow = slow.next
		fast = fast.next.next
		if slow == fast:
			return True
	return False

a = Node(1)
b = Node(2)
c = Node(3)

a.next = b
b.next = c
c.next = a

x = Node(1)
y = Node(2)
z = Node(3)

x.next = y
y.next = z


class Valid_Circular:
	def test(self, func):
		assert_equal(func(a), True)
		assert_equal(func(x), False)
	print("TESTS PASSED")

t = Valid_Circular()
t.test(valid_circular_LL)